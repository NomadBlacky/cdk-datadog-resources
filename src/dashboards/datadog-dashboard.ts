import { CfnResource, Construct } from '@aws-cdk/core';
import * as camelcaseKeys from 'camelcase-keys';
import { DatadogCredentials } from '../common/properties';

export interface DatadogDashboardProps {
  /** Credentials for the Datadog API */
  readonly datadogCredentials: DatadogCredentials;

  /** JSON string of the dashboard definition */
  readonly dashboardDefinition: string;
}

/**
 * Datadog Dashboard 1.0.0
 */
export class DatadogDashboard {
  constructor(scope: Construct, id: string, props: DatadogDashboardProps) {
    const cfnProperties = camelcaseKeys(props, {
      deep: true,
      pascalCase: true,
    });
    new CfnResource(scope, id, {
      type: 'Datadog::Dashboards::Dashboard',
      properties: { ...cfnProperties },
    });
  }
}