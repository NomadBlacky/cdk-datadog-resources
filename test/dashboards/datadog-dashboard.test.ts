import * as fs from 'fs';
import { SynthUtils } from '@aws-cdk/assert';
import { Stack } from '@aws-cdk/core';
import { DatadogDashboard } from '../../src/dashboards/datadog-dashboard';

test('Snapshot test', () => {
  const stack = new Stack();

  const dashboardDefJson = fs.readFileSync(`${__dirname}/dashboard-def.json`).toString();

  new DatadogDashboard(stack, 'TestMonitor', {
    datadogCredentials: {
      apiKey: 'DATADOG_API_KEY',
      applicationKey: 'DATADOG_APP_KEY',
    },
    dashboardDefinition: dashboardDefJson,
  });

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
