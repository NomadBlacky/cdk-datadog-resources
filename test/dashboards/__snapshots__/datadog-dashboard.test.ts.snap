// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "Resources": Object {
    "TestMonitor": Object {
      "Properties": Object {
        "DashboardDefinition": "{
  \\"title\\": \\"Test from cdk-datadog-resources\\",
  \\"description\\": \\"\\",
  \\"widgets\\": [
    {
      \\"id\\": 7963397977738346,
      \\"definition\\": {
        \\"title\\": \\"\\",
        \\"title_size\\": \\"16\\",
        \\"title_align\\": \\"left\\",
        \\"time\\": {

        },
        \\"type\\": \\"query_value\\",
        \\"requests\\": [
          {
            \\"q\\": \\"avg:datadog.estimated_usage.hosts{*}\\",
            \\"aggregator\\": \\"avg\\",
            \\"conditional_formats\\": [
              {
                \\"comparator\\": \\"<=\\",
                \\"palette\\": \\"white_on_yellow\\",
                \\"value\\": 0
              },
              {
                \\"comparator\\": \\">=\\",
                \\"palette\\": \\"white_on_green\\",
                \\"value\\": 1
              }
            ]
          }
        ],
        \\"autoscale\\": true,
        \\"precision\\": 2
      }
    }
  ],
  \\"template_variables\\": [

  ],
  \\"layout_type\\": \\"ordered\\",
  \\"is_read_only\\": false,
  \\"notify_list\\": [

  ],
  \\"id\\": \\"k87-rwz-htu\\"
}
",
        "DatadogCredentials": Object {
          "ApiKey": "DATADOG_API_KEY",
          "ApplicationKey": "DATADOG_APP_KEY",
        },
      },
      "Type": "Datadog::Dashboards::Dashboard",
    },
  },
}
`;
